2 - Templates/extra_scripts/shape_convert.R

```diff
@@ -28,77 +6,164 @@ pacman::p_load(
   dplyr
 )
 
-shape_convert <- function(old_shape, new_shape, weight, type) {
+#' Convert a measure from one geography to another
+#'
+#' This function takes a spatial dataset (`old_shape`) containing a measure
+#' and converts it into a new spatial dataset (`new_shape`).
+#' The conversion can be unweighted, or weighted based on area overlap, an
+#' external measure (e.g., population), or both.
+#'
+#' @param old_shape An `sf` object representing the original spatial dataset
+#' with measures. It must contain an `area_code` column and, if required, a
+#' weighting column.
+#' @param new_shape An `sf` object representing the target spatial dataset to
+#' which the measure should be converted. It must contain an `area_code` column.
+#' @param weight A character string specifying the weighting method:
+#'   - `"none"`: No weighting, taking a simple mean or sum.
+#'   - `"weighting"`: Weights the measure based on an external measure.
+#'   - `"overlap"`: Weights based on the percentage overlap between `old_shape`
+#'   and `new_shape`.
+#'   - `"weighting and overlap"`: Combines both external weighting and
+#'   percentage overlap.
+#' @param type A character string to indicate whether we want the new area
+#' measure to be calculated as mean or sum or the old area measure.
+
+shape_convert <- function(old_shape, new_shape, weight, type = "sum") {
+  
```

Comment:
validate input options (check type and weight are in specific options)

specify variables needed (area.code)